packages/ui/src/components/TagInput/index.tsx

```diff
@@ -220,48 +267,140 @@ export const TagInput = ({
     try {
       dispatchOnChange(newTagInput)
       setStatus({ [newTagInput.length - 1]: STATUS.IDLE })
-    } catch (error) {
-      onChangeError?.(error as Error)
+    } catch (err) {
       setTagInput(tagInputState)
     }
   }
 
+  const clearAll = () => {
+    setInput('')
+    setTagInput([])
+    dispatchOnChange([])
+  }
+
+  const helperSentiment = useMemo(() => {
+    if (error) {
+      return 'danger'
+    }
+
+    if (success) {
+      return 'success'
+    }
+
+    return 'neutral'
+  }, [error, success])
+
+  const computedClearable = clearable && !!tagInputState.length
+
   return (
-    <TagInputContainer
-      onClick={handleContainerClick}
-      variant={variant}
-      className={className}
-      data-testid={dataTestId}
-    >
-      {tagInputState.map(tag => (
-        <Tag
-          sentiment="neutral"
-          disabled={disabled}
-          key={tag.index}
-          isLoading={status[tag.index] === STATUS.LOADING}
-          onClose={e => {
-            e.stopPropagation()
-            deleteTag(tag.index)
-          }}
+    <Stack gap="0.5" className={className}>
+      <Stack direction="row" gap="1" alignItems="center">
+        <Stack direction="row" gap="0.5" alignItems="start">
+          <Text
+            as="label"
+            variant="bodyStrong"
+            sentiment="neutral"
+            htmlFor={id ?? localId}
+          >
+            {label}
+          </Text>
+          {required ? <Icon name="asterisk" color="danger" size={8} /> : null}
+        </Stack>
+        {labelDescription ?? null}
+      </Stack>
+      <div>
+        <Tooltip text={tooltip}>
+          <TagInputContainer
+            onClick={handleContainerClick}
+            className={className}
+            data-testid={dataTestId}
+            size={size}
+            data-disabled={disabled}
+            data-readonly={readOnly}
+            data-error={!!error}
+            data-success={!!success}
+          >
+            <DataContainer>
+              {tagInputState.map(tag => (
+                <Tag
+                  sentiment="neutral"
+                  disabled={disabled}
+                  key={tag.index}
+                  isLoading={status[tag.index] === STATUS.LOADING}
+                  onClose={
+                    !readOnly
+                      ? e => {
+                          e.stopPropagation()
+                          deleteTag(tag.index)
+                        }
+                      : undefined
+                  }
+                >
+                  {tag.label}
+                </Tag>
+              ))}
+              {!disabled ? (
+                <StyledInput
+                  id={id}
```

Comment:
Here you should pass `id ?? localId` or it won't work when we click on label

```suggestion
                  id={id ?? localId}
```